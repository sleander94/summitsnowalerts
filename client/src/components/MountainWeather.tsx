import { useEffect, useState } from 'react';
import { WeatherData, WeatherProps } from '../types.d';

const MountainWeather = ({ name, location, completeLoading }: WeatherProps) => {
  const [weather, setWeather] = useState<WeatherData>();
  const [content, setContent] = useState<boolean>(false);

  const getConditions = async (location: number) => {
    const response = await fetch(
      `https://api.weatherapi.com/v1/forecast.json?key=${process.env.REACT_APP_WEATHER_KEY}=${location}&days=2&aqi=no&alerts=no`
    );
    const data = await response.json();
    setWeather({
      today: {
        condition: data.forecast.forecastday[0].day.condition.text,
        icon: data.forecast.forecastday[0].day.condition.icon,
        avgtemp_f: data.forecast.forecastday[0].day.avgtemp_f,
        daily_chance_of_snow:
          data.forecast.forecastday[0].day.daily_chance_of_snow,
        totalprecip_in: data.forecast.forecastday[0].day.totalprecip_in,
      },
      tomorrow: {
        condition: data.forecast.forecastday[1].day.condition.text,
        icon: data.forecast.forecastday[1].day.condition.icon,
        avgtemp_f: data.forecast.forecastday[1].day.avgtemp_f,
        daily_chance_of_snow:
          data.forecast.forecastday[1].day.daily_chance_of_snow,
        totalprecip_in: data.forecast.forecastday[1].day.totalprecip_in,
      },
    });
    setContent(true);
    completeLoading();
  };

  useEffect(() => {
    getConditions(location);
  }, [location]);

  return (
    <div id={name} className="mountain-weather">
      {content && (
        <div className="weather-content">
          <h2>{name}</h2>
          <div className="current-conditions">
            <h3>Today</h3>
            <p>{weather?.today.condition}</p>
            <img
              src={weather?.today.icon}
              alt={`${weather?.today.condition} icon`}
            />
            <p>{weather?.today.avgtemp_f} °F</p>
            <p>Snow: {weather?.today.daily_chance_of_snow}%</p>
            <p>Precip: {weather?.today.totalprecip_in} in.</p>
          </div>
          <div className="forecast">
            <h3>Tomorrow</h3>
            <p>{weather?.tomorrow.condition}</p>
            <img
              src={weather?.tomorrow.icon}
              alt={`${weather?.tomorrow.condition} icon`}
            />
            <p>{weather?.tomorrow.avgtemp_f} °F</p>
            <p>Snow: {weather?.tomorrow.daily_chance_of_snow}%</p>
            <p>Precip: {weather?.tomorrow.totalprecip_in} in.</p>
          </div>
        </div>
      )}
    </div>
  );
};

export default MountainWeather;
