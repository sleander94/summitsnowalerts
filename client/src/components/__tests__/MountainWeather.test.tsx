import MountainWeather from '../MountainWeather';
import { render, screen, waitFor } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import { rest } from 'msw';
import { setupServer } from 'msw/node';

const server = setupServer(
  rest.get('https://api.weatherapi.com/v1/forecast.json', (req, res, ctx) => {
    return res(
      ctx.json({
        forecast: {
          forecastday: [
            {
              day: {
                condition: {
                  text: 'Sunny',
                  icon: 'fakeurl',
                },
                avgtemp_f: 72.3,
                daily_chance_of_snow: 0,
                totalprecip_in: 0,
              },
            },
            {
              day: {
                condition: {
                  text: 'Snowy',
                  icon: 'fakeurl',
                },
                avgtemp_f: 28,
                daily_chance_of_snow: 0.95,
                totalprecip_in: 5.5,
              },
            },
          ],
        },
      })
    );
  })
);

beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());

it('Renders properly after loading data', async () => {
  const component = render(
    <Router>
      <MountainWeather
        name="Test Location"
        location={12345}
        completeLoading={() => {}}
      />
    </Router>
  );

  await waitFor(() =>
    expect(
      screen.getByRole('heading', { name: 'Test Location' })
    ).toBeInTheDocument()
  );

  expect(component).toMatchSnapshot();
});
