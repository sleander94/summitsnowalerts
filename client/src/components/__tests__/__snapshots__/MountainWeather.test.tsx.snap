// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders properly after loading data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mountain-weather"
        id="Test Location"
      >
        <div
          class="weather-content"
        >
          <h2>
            Test Location
          </h2>
          <div
            class="current-conditions"
          >
            <h3>
              Today
            </h3>
            <p>
              Sunny
            </p>
            <img
              alt="Sunny icon"
              src="fakeurl"
            />
            <p>
              72.3
               째F
            </p>
            <p>
              Snow: 
              0
              %
            </p>
            <p>
              Precip: 
              0
               in.
            </p>
          </div>
          <div
            class="forecast"
          >
            <h3>
              Tomorrow
            </h3>
            <p>
              Snowy
            </p>
            <img
              alt="Snowy icon"
              src="fakeurl"
            />
            <p>
              28
               째F
            </p>
            <p>
              Snow: 
              95
              %
            </p>
            <p>
              Precip: 
              5.5
               in.
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mountain-weather"
      id="Test Location"
    >
      <div
        class="weather-content"
      >
        <h2>
          Test Location
        </h2>
        <div
          class="current-conditions"
        >
          <h3>
            Today
          </h3>
          <p>
            Sunny
          </p>
          <img
            alt="Sunny icon"
            src="fakeurl"
          />
          <p>
            72.3
             째F
          </p>
          <p>
            Snow: 
            0
            %
          </p>
          <p>
            Precip: 
            0
             in.
          </p>
        </div>
        <div
          class="forecast"
        >
          <h3>
            Tomorrow
          </h3>
          <p>
            Snowy
          </p>
          <img
            alt="Snowy icon"
            src="fakeurl"
          />
          <p>
            28
             째F
          </p>
          <p>
            Snow: 
            95
            %
          </p>
          <p>
            Precip: 
            5.5
             in.
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
